### ISSUE TITLE
loss becomes nan during training

### ISSUE THREAD

**Issue by Davidwhw** [Vision-CAIR/MiniGPT-4#517]
When I pre-trained for phase 1 using the coco dataset (downloaded using the [script](https://github.com/salesforce/LAVIS/blob/main/lavis/datasets/download_scripts/download_coco.py) in lavis), loss quickly became nan. This issue did not occur when I used this dataset to fine-tune BLIP2. 

I only changed the `datasets` part and left the rest as the default Settings:
*train_configs/minigpt4_stage1_pretrain.yaml*：
> model:
>   arch: minigpt4
>   model_type: pretrain_vicuna0
> 
> datasets:
>   coco_caption:
>     batch_size: 64
>     vis_processor:
>         train:
>           name: "blip2_image_train"
>           image_size: 224
>     text_processor:
>         train:
>           name: "blip_caption"
>     sample_ratio: 115
> 
> run:
>   task: image_text_pretrain
>   #optimizer
>   lr_sched: "linear_warmup_cosine_lr"
>   init_lr: 1e-4
>   min_lr: 8e-5
>   warmup_lr: 1e-6
> 
>   weight_decay: 0.05
>   max_epoch: 4
>   num_workers: 4
>   warmup_steps: 5000
>   iters_per_epoch: 5000
> 
>   seed: 42
>   output_dir: "output/minigpt4_stage1_pretrain"
> 
>   amp: True
>   resume_ckpt_path: null
> 
>   evaluate: False 
>   train_splits: ["train"]
> 
>   device: "cuda"
>   world_size: 1
>   dist_url: "env://"
>   distributed: True
> 
>   wandb_log: False
>   job_name: minigpt4_pretrain

Here is the training log:

> 2024-06-12 16:20:56,611 [INFO] Start training
2024-06-12 16:21:06,082 [INFO] dataset_ratios not specified, datasets will be concatenated (map-style datasets) or chained (webdataset.DataPipeline).
2024-06-12 16:21:06,082 [INFO] Loaded 414113 records for train split from the dataset.
batch sizes [[64]]
module.llama_proj.weight
module.llama_proj.bias
2024-06-12 16:21:06,100 [INFO] number of trainable parameters: 3149824
2024-06-12 16:21:06,101 [INFO] Start training epoch 0, 5000 iters per inner epoch.
[W reducer.cpp:1300] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration,  which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator())
[W reducer.cpp:1300] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration,  which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator())
Train: data epoch: [0]  [   0/5000]  eta: 6:21:59  lr: 0.000000  loss: 6.8750  time: 4.5839  data: 0.0000  max mem: 54099
Train: data epoch: [0]  [  50/5000]  eta: 5:30:41  lr: 0.000000  loss: 6.9350  time: 4.3366  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [ 100/5000]  eta: 5:36:02  lr: 0.000000  loss: nan  time: 4.2825  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [ 150/5000]  eta: 5:35:11  lr: 0.000000  loss: 6.9082  time: 4.2402  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [ 200/5000]  eta: 5:32:37  lr: 0.000000  loss: 6.8793  time: 4.1392  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [ 250/5000]  eta: 5:29:55  lr: 0.000001  loss: nan  time: 4.0571  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [ 300/5000]  eta: 5:19:55  lr: 0.000001  loss: nan  time: 3.5617  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [ 350/5000]  eta: 5:11:46  lr: 0.000001  loss: nan  time: 3.4891  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [ 400/5000]  eta: 5:03:33  lr: 0.000001  loss: nan  time: 3.5205  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [ 450/5000]  eta: 4:57:14  lr: 0.000001  loss: nan  time: 3.5452  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [ 500/5000]  eta: 4:51:30  lr: 0.000001  loss: nan  time: 3.4910  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [ 550/5000]  eta: 4:46:23  lr: 0.000001  loss: nan  time: 3.4316  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [ 600/5000]  eta: 4:41:38  lr: 0.000001  loss: nan  time: 3.5642  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [ 650/5000]  eta: 4:37:05  lr: 0.000001  loss: nan  time: 3.5331  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [ 700/5000]  eta: 4:33:05  lr: 0.000001  loss: nan  time: 3.4861  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [ 750/5000]  eta: 4:29:03  lr: 0.000002  loss: nan  time: 3.4726  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [ 800/5000]  eta: 4:25:02  lr: 0.000002  loss: nan  time: 3.6616  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [ 850/5000]  eta: 4:21:01  lr: 0.000002  loss: nan  time: 3.5413  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [ 900/5000]  eta: 4:17:19  lr: 0.000002  loss: nan  time: 3.6708  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [ 950/5000]  eta: 4:13:42  lr: 0.000002  loss: nan  time: 3.6225  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [1000/5000]  eta: 4:10:01  lr: 0.000002  loss: nan  time: 3.5988  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [1050/5000]  eta: 4:06:16  lr: 0.000002  loss: nan  time: 3.5804  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [1100/5000]  eta: 4:02:48  lr: 0.000002  loss: nan  time: 3.5840  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [1150/5000]  eta: 3:59:17  lr: 0.000002  loss: nan  time: 3.6335  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [1200/5000]  eta: 3:55:56  lr: 0.000002  loss: nan  time: 3.5396  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [1250/5000]  eta: 3:52:35  lr: 0.000003  loss: nan  time: 3.6103  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [1300/5000]  eta: 3:49:22  lr: 0.000003  loss: nan  time: 3.5434  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [1350/5000]  eta: 3:46:05  lr: 0.000003  loss: nan  time: 3.6506  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [1400/5000]  eta: 3:42:52  lr: 0.000003  loss: nan  time: 3.4917  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [1450/5000]  eta: 3:39:39  lr: 0.000003  loss: nan  time: 3.7636  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [1500/5000]  eta: 3:36:24  lr: 0.000003  loss: nan  time: 3.4870  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [1550/5000]  eta: 3:33:07  lr: 0.000003  loss: nan  time: 3.6956  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [1600/5000]  eta: 3:29:53  lr: 0.000003  loss: nan  time: 3.5675  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [1650/5000]  eta: 3:26:49  lr: 0.000003  loss: nan  time: 3.8908  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [1700/5000]  eta: 3:23:30  lr: 0.000003  loss: nan  time: 3.6449  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [1750/5000]  eta: 3:20:18  lr: 0.000004  loss: nan  time: 3.6010  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [1800/5000]  eta: 3:17:11  lr: 0.000004  loss: nan  time: 3.7217  data: 0.0000  max mem: 55634
Train: data epoch: [0]  [1850/5000]  eta: 3:13:58  lr: 0.000004  loss: nan  time: 3.6164  data: 0.0000  max mem: 55634
...

Can you give me some advice or clues?
Thank you for your assistance.
